src/utils/transclude-directive.js

```diff
@@ -0,0 +1,170 @@
+ * @restrict A
+ * @element ANY
+ * @param {string} pfTransclude specifies the type of transclusion to use.<br/>
+ * <strong>Values:</strong>
+ * <ul style='list-style-type: none'>
+ * <li> 'sibling' - The transcluded contents scope is a sibling one to the element where transclusion happens (default)
+ * <li> 'parent'  - The transcluded contents scope is that of the element where transclusion happens.
+ * <li> 'child'   - The transcluded contents scope is child scope to the scope of the element where transclusion happens.
+ * </ul>
+ *
+ * @description
+ *   Directive for transcluding in directives and setting up scope of children of parent directives. This is a workaround
+ *   for https://github.com/angular/angular.js/issues/5489
+ *
+ * @example
+<example module="patternfly.utils">
+  <file name="index.html">
+    <style>
+
+      .pf-transclude-example div {
+        border: 1px solid #337ab7;
+        margin-bottom: 20px;
+        margin-left:   20px;
+      }
+
+      .pf-transclude-example p {
+        background-color: #337ab7;
+        margin: 0;
+        padding: 5px 10px;
+      }
+
+     .pf-transclude-example id {
+        display: inline-block;
+        background-color: #def3ff;
+        color: #000000;
+        border-radius: 10px;
+        width: 20px;
+        height: 20px;
+        text-align: center;
+        line-height: 20px;
+        margin-left: 5px;
+      }
+
+      .pf-transclude-example pre {
+        padding: 5px;
+        border-width: 0px;
+      }
+    </style>
+    <div ng-controller="UtilCtrl" class="row pf-transclude-example" style="display:inline-block; width: 100%;">
+      <span>Here the scope id is: <id>{{$id}}</id></span>
+
+      <transclude-sibling class="pf-transclude-example">
+        <pre>This content was transcluded using <b>pf-transclude</b> or <b>pf-transclude="sibling"</b>.</pre><pre>Its scope is: <id>{{$id}}</id> the parent of which is <id>{{$parent.$id}}</id></pre>
+      </transclude-sibling>
+
+      <transclude-parent>
+        <pre>This content was transcluded using <b>pf-transclude="parent"</b>.</pre><pre>Its scope is: <id>{{$id}}</id> the parent of which is <id>{{$parent.$id}}</id></pre>
+      </transclude-parent>
+
+      <transclude-child>
+        <pre>This content was transcluded using <b>pf-transclude="child"</b>.</pre><pre>Its scope is: <id>{{$id}}</id> the parent of which is <id>{{$parent.$id}}</id></pre>
+      </transclude-child>
+    </div>
+  </file>
+
+  <file name="script.js">
+    angular.module('patternfly.utils')
+      .controller( 'UtilCtrl', function($scope) {
+
+      })
+
+      .config(function($provide){
+          $provide.decorator('ngTranscludeDirective', ['$delegate', function($delegate) {
+              // Remove the original directive
+              $delegate.shift();
+              return $delegate;
+          }]);
+      })
+
+      .directive( 'transcludeSibling', function() {
+        return {
+          restrict: 'E',
+          transclude: true,
```

Comment:
Any reason why these are restricted to just 'E'? normal transcludes support both